evadb/optimizer/optimizer_utils.py

```diff
@@ -206,26 +241,21 @@ def optimize_cache_key(context: "OptimizerContext", expr: FunctionExpression):
 
     """
     keys = expr.children
-    catalog = context.db.catalog()
-    # handle simple one column inputs
+
+    # Handle simple one column inputs.
     if len(keys) == 1 and isinstance(keys[0], TupleValueExpression):
-        child = keys[0]
-        col_catalog_obj = child.col_object
-        if isinstance(col_catalog_obj, ColumnCatalogEntry):
-            new_keys = []
-            table_obj = catalog.get_table_catalog_entry(col_catalog_obj.table_name)
-            for col in get_table_primary_columns(table_obj):
-                new_obj = catalog.get_column_catalog_entry(table_obj, col.name)
-                new_keys.append(
-                    TupleValueExpression(
-                        name=col.name,
-                        table_alias=child.table_alias,
-                        col_object=new_obj,
-                        col_alias=f"{child.table_alias}.{col.name}",
-                    )
-                )
+        return optimize_cache_key_for_tuple_value_expression(context, keys[0])
+
+    # Handle ConstantValueExpressin + TupleValueExpression
+    if (
+        len(keys) == 2
+        and isinstance(keys[0], ConstantValueExpression)
+        and isinstance(keys[1], TupleValueExpression)
+    ):
```

Comment:
Instead of hardcoding, can do register a mapping between the `optimize_cache_key_for ...` and `Expressions`. For example: 
```
optimize_cache_key_registry = {
    ConstantValueExpression: 'optimize_cache_key_for_constant_value_expression', 
    TupleValueExpression: 'optimize_cache_key_for_tuple_value_expression',
}
```
And we can use a for loop to iterate over all keys.

Reply:
Even I wanted it. But I guess we can do it in a later PR.

Reply:
Fixed this.